src/utils/memo.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * For running something once
+ * @example
+ * const getProvider = once((ctx) => new R2Provider({ ctx }));
+ * // later...
+ * getProvider(ctx).something();
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Use a `I` to specific type of `Array<I>` and make it `unknown`

Reply:
Like `<T, Args extends Array<I>, I = unknown>` ?

Reply:
Possibly